src/compiler/crystal/macros.cr

```diff
@@ -874,9 +874,17 @@ module Crystal::Macros
 
     # Returns the value of a named argument,
     # or NilLiteral if the named argument isn't
-    # used in this attribute.
+    # used on `self`.
     def [](name : SymbolLiteral | StringLiteral | MacroId) : ASTNode
     end
+
+    # Returns a `NamedTupleLiteral` representing the named arguments on `self`.
+    def named_args : NamedTupleLiteral
+    end
+
+    # Returns a `TupleLiteral` representing the positional arguments on `self`.
+    def pos_args : TupleLiteral
```

Comment:
I would prefer to name them `args`, like in `Call` `ASTNode`.

Reply:
How would you differentiate between the positional/named args then?